src/globus_sdk/services/transfer/client.py

```diff
@@ -2699,3 +2699,195 @@ def endpoint_manager_delete_pause_rule(
             f"/v0.10/endpoint_manager/pause_rule/{pause_rule_id}",
             query_params=query_params,
         )
+
+    # Tunnel methods
+
+    def create_tunnel(
+        self,
+        data: dict[str, t.Any] | CreateTunnelData,
+    ) -> response.GlobusHTTPResponse:
+        """
+        :param data: Parameters for the tunnel creation
+
+        .. tab-set::
+
+            .. tab-item:: Example Usage
+
+                .. code-block:: python
+
+                    tc = globus_sdk.TunnelClient(...)
+                    result = tc.create_tunnel(data)
+                    print(result["data"]["id"])
+
+            .. tab-item:: API Info
+
+                ``POST /v2/tunnels``
+        """
+        log.debug("TransferClient.create_tunnel(...)")
+        try:
+            data_element = data["data"]
+        except KeyError:
+            raise exc.GlobusSDKUsageError(
+                "create_tunnel() body was malformed (missing the 'data' key). "
+                "Use CreateTunnelData to easily create correct documents."
+            )
```

Comment:
pylint is mad about this right now, since it's implicit exception chaining. It wants us to be explicit:

```suggestion
        except KeyError as e:
            raise exc.GlobusSDKUsageError(
                "create_tunnel() body was malformed (missing the 'data' key). "
                "Use CreateTunnelData to easily create correct documents."
            ) from e
```

(Aside: we don't run pylint via pre-commit because it's so gosh-darn slow.)